lucene/sandbox/src/java/org/apache/lucene/sandbox/facet/plain/histograms/HistogramCollector.java

```diff
@@ -330,4 +333,23 @@ static void checkMaxBuckets(int size, int maxBuckets) {
   public void setWeight(Weight weight) {
     this.weight = weight;
   }
+
+  private boolean isMatchAll(LeafReaderContext context) throws IOException {
+    return weight != null && weight.count(context) == context.reader().maxDoc();
+  }
+
+  private PointRangeQuery getPointRangeQuery(final String field) {
+    if (weight == null || weight.getQuery() == null) {
+      return null;
+    }
+
+    if (weight.getQuery() instanceof PointRangeQuery) {
```

Comment:
Nit: you could use more modern syntax since `main` requires Java 21 now.

```suggestion
    if (weight.getQuery() instanceof PointRangeQuery prq) {
```

Reply:
Thanks for sharing this. Really nice improvement, wasn't aware of it. `instanceof` is bit of a pain otherwise!